src/studio/src/designer/frontend/packages/schema-editor/src/components/SchemaInspector/ReferenceSelectionComponent.tsx

```diff
@@ -1,36 +1,39 @@
 import React from 'react';
-import { ClassNameMap } from '@material-ui/core/styles/withStyles';
-import { RefSelect } from './RefSelect';
 import type { UiSchemaNode } from '@altinn/schema-model';
+import { Select } from '../common/Select';
+import { getDomFriendlyID } from '../../utils/ui-schema-utils';
+import { useSelector } from 'react-redux';
+import { ISchemaState } from '../../types';
+import { getRootNodes } from '@altinn/schema-model';
+import classes from './ReferenceSelectionComponent.module.css';
 
 export interface IReferenceSelectionProps {
   buttonText: string;
-  classes: ClassNameMap;
+  emptyOptionLabel: string;
   label: string;
   onChangeRef: (refPointer: string, value: string) => void;
   onGoToDefButtonClick: () => void;
   selectedNode: UiSchemaNode;
 }
 
 export function ReferenceSelectionComponent({
-  classes,
+  emptyOptionLabel,
   selectedNode,
   label,
   buttonText,
   onChangeRef,
   onGoToDefButtonClick,
 }: IReferenceSelectionProps) {
-  const common = {
-    label,
-    nodePointer: selectedNode.pointer,
-    fullWidth: true,
-  };
+  const definitions: UiSchemaNode[] = useSelector((state: ISchemaState) => getRootNodes(state.uiSchema, true));
   return (
     <div>
-      <RefSelect
-        {...common}
-        value={selectedNode.ref ?? ''}
+      <Select
+        emptyOptionLabel={emptyOptionLabel}
+        id={getDomFriendlyID(selectedNode.pointer, 'ref-select')}
+        label={label}
         onChange={(value) => onChangeRef(selectedNode.pointer, value)}
+        options={definitions.map(({pointer}) => ({ value: pointer, label: pointer }))}
+        value={selectedNode.ref?.replace('#/definitions/', '') || ''}
```

Comment:
`#/definitions/` burde vel kanskje vært `$defs` ... men jeg ville brukt `makePointer(Keywords.Definition)`, fordelen med dette er at det vil være kjempelett å kunne se hvilken kode som er avhengig av dette keywordet.

Reply:
Tror egentlig ikke denne `replace` er nødvendig i det hele tatt. Dette er hentet rett fra den gamle koden, men jeg forstår ikke helt hva det skal være godt for.